File: Problems/No.1291_Sequential_Digits/sequential_digits.py
class Solution(object):
    def sequentialDigits(self, low, high):
        """
        :type low: int
        :type high: int
        :rtype: List[int]
        """

        # parameter
        sequential_digits = []

        # start computing
        digits = "123456789"
        for i in range(len(str(low)), len(str(high))+1):
            for idx in range(9-i+1):
                temp_digit = int(digits[idx:idx+i])
                if (low <= temp_digit <= high):
                    sequential_digits.append(temp_digit)
                elif (temp_digit < low):
                    continue
                else:
                    break
        
        return sequential_digits

File: Problems/No.1291_Sequential_Digits/sequential_digits.cpp
class Solution {
public:
    vector<int> sequentialDigits(int low, int high) {
        
        // parameter
        std::vector<int> sequential_digits;

        // start computing
        std::string digits = "123456789";
        for (int iter=std::to_string(low).length(); iter<=std::to_string(high).length(); iter++) 
            for (int idx=0; idx<=9-iter; idx++) {

                int temp_digit = stoi(digits.substr(idx, iter));
                if ((low <= temp_digit) && (temp_digit <= high)) sequential_digits.push_back(temp_digit);
                else if (temp_digit < low) continue;
                else break;

            }

        return sequential_digits;

    }
};
